lib/pymedphys/_streamlit/utilities/session.py

```diff
@@ -21,10 +21,17 @@
 
 
 class SessionState:
-    def __init__(self, **kwargs):
-        for key, val in kwargs.items():
+    def __init__(self, state):
+        for key, val in state.items():
             setattr(self, key, val)
 
+    def update(self, state):
+        for key, val in state.items():
+            try:
```

Comment:
isn't there a has_key so that you can use a conditional instead of an exception?

Reply:
For here, EAFP instead of LBYL tends to be the Python idiom. See https://devblogs.microsoft.com/python/idiomatic-python-eafp-versus-lbyl/

Reply:
I stand corrected.
Misplaced habit.

Reply:
The docs have some neat comments on EAFP regarding multi-threaded environments:

https://docs.python.org/3.5/glossary.html#term-lbyl